From 4e01157aded3b9e86ddf4a9e863ed09348c9f7d1 Mon Sep 17 00:00:00 2001
From: Tomasz Maciej Nowak <tomek_n@o2.pl>
Date: Wed, 14 Feb 2018 17:34:38 +0100
Subject: [PATCH 4/9] mvebu: unify boot.scr creation

Unify boot.scr generation so Makefile for device image generation won't
grow without a reason. Also make boot-scr step optional.

Signed-off-by: Tomasz Maciej Nowak <tomek_n@o2.pl>
---
 target/linux/mvebu/image/Makefile                       | 17 ++++++-----------
 ...-boot.script => armada-388-clearfog-base.bootscript} |  2 +-
 ...o-boot.script => armada-388-clearfog-pro.bootscript} |  2 +-
 3 files changed, 8 insertions(+), 13 deletions(-)
 rename target/linux/mvebu/image/{cfbase-boot.script => armada-388-clearfog-base.bootscript} (94%)
 rename target/linux/mvebu/image/{cfpro-boot.script => armada-388-clearfog-pro.bootscript} (93%)

diff --git a/target/linux/mvebu/image/Makefile b/target/linux/mvebu/image/Makefile
index 3ea9775bcb..df07ae108f 100644
--- a/target/linux/mvebu/image/Makefile
+++ b/target/linux/mvebu/image/Makefile
@@ -25,14 +25,9 @@ endef
 # Partition 1:   32768 sectors
 # Partition 2:   98304 sectors (configurable)
 
-define Build/boot-scr-cfpro
-	rm -f $@.bootscript
-	mkimage -A arm -O linux -T script -C none -a 0 -e 0 -d cfpro-boot.script $@.bootscript
-endef
-
-define Build/boot-scr-cfbase
-	rm -f $@.bootscript
-	mkimage -A arm -O linux -T script -C none -a 0 -e 0 -d cfbase-boot.script $@.bootscript
+define Build/boot-scr
+	rm -f $@-boot.scr
+	mkimage -A arm -O linux -T script -C none -a 0 -e 0 -d $(DEVICE_NAME).bootscript $@-boot.scr
 endef
 
 define Build/boot-img
@@ -40,7 +35,7 @@ define Build/boot-img
 	mkfs.fat -C $@.boot 16384
 	$(foreach dts,$(DEVICE_DTS), mcopy -i $@.boot $(DTS_DIR)/$(dts).dtb ::$(dts).dtb;)
 	mcopy -i $@.boot $(IMAGE_KERNEL) ::zImage
-	mcopy -i $@.boot $@.bootscript ::boot.scr
+	-mcopy -i $@.boot $@-boot.scr ::boot.scr
 endef
 
 define Build/sdcard-img
@@ -200,7 +195,7 @@ define Device/armada-388-clearfog-pro
   DEVICE_TITLE := SolidRun ClearFog Pro
   DEVICE_PACKAGES := mkf2fs e2fsprogs swconfig kmod-fs-vfat kmod-nls-cp437 kmod-nls-iso8859-1
   IMAGES := sdcard.img.gz
-  IMAGE/sdcard.img.gz := boot-scr-cfpro | boot-img | sdcard-img | gzip | append-metadata
+  IMAGE/sdcard.img.gz := boot-scr | boot-img | sdcard-img | gzip | append-metadata
   DEVICE_DTS := armada-388-clearfog-pro armada-388-clearfog-base
   SUPPORTED_DEVICES := armada-388-clearfog-pro armada-388-clearfog
   UBOOT := clearfog-u-boot-spl.kwb
@@ -213,7 +208,7 @@ define Device/armada-388-clearfog-base
   DEVICE_TITLE := SolidRun ClearFog Base
   DEVICE_PACKAGES := mkf2fs e2fsprogs kmod-fs-vfat kmod-nls-cp437 kmod-nls-iso8859-1
   IMAGES := sdcard.img.gz
-  IMAGE/sdcard.img.gz := boot-scr-cfbase | boot-img | sdcard-img | gzip | append-metadata
+  IMAGE/sdcard.img.gz := boot-scr | boot-img | sdcard-img | gzip | append-metadata
   DEVICE_DTS := armada-388-clearfog-pro armada-388-clearfog-base
   UBOOT := clearfog-u-boot-spl.kwb
 endef
diff --git a/target/linux/mvebu/image/cfbase-boot.script b/target/linux/mvebu/image/armada-388-clearfog-base.bootscript
similarity index 94%
rename from target/linux/mvebu/image/cfbase-boot.script
rename to target/linux/mvebu/image/armada-388-clearfog-base.bootscript
index 89fa1dc578..663cb581cc 100644
--- a/target/linux/mvebu/image/cfbase-boot.script
+++ b/target/linux/mvebu/image/armada-388-clearfog-base.bootscript
@@ -1,6 +1,6 @@
 # Standard Boot-Script
 # use only well-known variable names provided by U-Boot Distro boot
-# This script assumes that there is a boot partition, 
+# This script assumes that there is a boot partition,
 # and that the root partition is always the next one.
 
 # Override DeviceTree for Clearfog Base
diff --git a/target/linux/mvebu/image/cfpro-boot.script b/target/linux/mvebu/image/armada-388-clearfog-pro.bootscript
similarity index 93%
rename from target/linux/mvebu/image/cfpro-boot.script
rename to target/linux/mvebu/image/armada-388-clearfog-pro.bootscript
index 1588c1546e..d2f153f686 100644
--- a/target/linux/mvebu/image/cfpro-boot.script
+++ b/target/linux/mvebu/image/armada-388-clearfog-pro.bootscript
@@ -1,6 +1,6 @@
 # Standard Boot-Script
 # use only well-known variable names provided by U-Boot Distro boot
-# This script assumes that there is a boot partition, 
+# This script assumes that there is a boot partition,
 # and that the root partition is always the next one.
 
 # generate bootargs for rootfs on MMC
-- 
2.16.2

